include/seqan3/alphabet/nucleotide/rna4.hpp

```diff
@@ -0,0 +1,277 @@
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+// Author: Chenxu Pan <chenxu.pan@fu-berlin.de>
+// ============================================================================
+
+#pragma once
+
+#include <cassert>
+
+#include "../alphabet.hpp"
+
+/*! The four letter DNA alphabet
+ * \ingroup alphabet
+ */
+
+namespace seqan3
+{
+
+/*! The four letter DNA alphabet of A,C,G,T
+ *
+ * The alphabet may be brace initialized from the static letter members (see above). Note that you cannot assign
+ * regular characters, but additional functions for this are available.
+ *
+ *~~~~~~~~~~~~~~~{.cpp}
+ *     rna4 my_letter{rna4::A};
+ *     // doesn't work:
+ *     // rna4 my_letter{'A'};
+ *
+ *     my_letter.from_char('C'); // <- this does!
+ *
+ *     my_letter.from_char('F'); // converted to A internally
+ *     if (my_letter.to_char() == 'A')
+ *        std::cout << "yeah\n"; // "yeah";
+ *~~~~~~~~~~~~~~~
+ */
+
+struct rna4
+{
+    //! the type of the alphabet when converted to char (e.g. via @link to_char @endlink)
+    using char_type = char;
+    //! the type of the alphabet when represented as a number (e.g. via @link to_integral @endlink)
+    using integral_type = uint8_t;
+
+    // strictly typed enum, unfortunately with scope
+    //! \privatesection
+    enum struct internal_type : integral_type
+    {
+        A,
+        C,
+        G,
+        U,
+        T = U,
+        UNKNOWN = A
+    };
+    internal_type value;
```

Comment:
we changed this, as well, so "private" members should now get an `_` to the beginning of the name in aggregate pod classes.